### ISSUE TITLE
feat(api): add interrater reliability computation engine

### ISSUE THREAD

**Issue by dtsong** [dtsong/biblical-evals#38]
## Overview

Add an interrater reliability (IRR) computation engine using the `krippendorff` pip package. Computes Krippendorff'"'"'s alpha, ICC, and pairwise weighted kappa to measure scoring consistency across expert panels. Per Strategist: use established packages, not custom implementations.

### Metrics

| Metric | Purpose | Package |
|---|---|---|
| **Krippendorff'"'"'s alpha** (ordinal) | Primary IRR metric, handles missing data | `krippendorff` |
| **ICC(2,k)** per dimension | Two-way random, absolute agreement | `scipy` / custom |
| **Pairwise weighted kappa** | Diagnostic, identifies specific rater disagreements | `sklearn.metrics` |

### New Endpoint

`GET /api/v1/studies/{id}/reliability` returning:

```json
{
  "overall_alpha": 0.78,
  "per_dimension": {"factual_accuracy": {"alpha": 0.85, "icc": 0.88}},
  "pairwise_kappa": {"expert_1:expert_2": {"factual_accuracy": 0.71}},
  "flagged_disagreements": [{"response_id": "...", "dimension": "...", "range": 4}],
  "reliability_status": "acceptable"
}
```

### Reliability Thresholds (per Scout'"'"'s research)

| Alpha | Status | Action |
|---|---|---|
| >= 0.800 | `reliable` | Proceed with analysis |
| 0.667-0.800 | `tentative` (acceptable) | Note in report, may proceed |
| < 0.667 | `poor` | Flag for recalibration |

### Disagreement Flagging

Flag any response-dimension pair where expert scores differ by >= 3 points (range >= 4 on a 1-5 scale).

## Files to Create/Modify

- [ ] `apps/api/src/scoring/reliability.py` — IRR computation module
- [ ] `apps/api/src/api/studies.py` — Add reliability endpoint
- [ ] `apps/api/pyproject.toml` — Add `krippendorff` dependency
- [ ] `apps/api/tests/test_reliability.py` — Tests with known IRR test cases

## Dependencies

- Blocked by #34 (StudyExpert model)
- Blocked by #35 (Expert panel with multi-rater data)

## Acceptance Criteria

- [ ] Krippendorff'"'"'s alpha computed correctly for ordinal data
- [ ] ICC(2,k) computed per dimension
- [ ] Pairwise kappa identifies rater disagreements
- [ ] Reliability status classified per thresholds
- [ ] Disagreements flagged (range >= 4)
- [ ] Handles missing data gracefully (experts who haven'"'"'t scored all responses)
- [ ] Tests pass with known IRR reference values

## Context

This is part of the Research-Grade Evaluation Roadmap (Phase 6). IRR metrics are essential for academic credibility — without them, multi-expert scoring is anecdotal rather than scientific.
